lms/static/js/student_account/components/StudentAccountDeletionModal.jsx

```diff
@@ -0,0 +1,212 @@
+/* globals gettext */
+/* eslint-disable react/no-danger */
+import React from 'react';
+import 'whatwg-fetch';
+import PropTypes from 'prop-types';
+import { Button, Modal, Icon, InputText, StatusAlert } from '@edx/paragon/static';
+import StringUtils from 'edx-ui-toolkit/js/utils/string-utils';
+
+class StudentAccountDeletionConfirmationModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.closeConfirmationModal = this.closeConfirmationModal.bind(this);
+    this.deleteAccount = this.deleteAccount.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.passwordFieldValidation = this.passwordFieldValidation.bind(this);
+    this.state = {
+      password: '',
+      passwordSubmitted: false,
+      isValid: true,
+      validationMessage: '',
+      accountQueuedForDeletion: false,
+      responseError: false,
+      responseErrorBody: {},
+    };
+  }
+
+  closeConfirmationModal() {
+    const { onClose } = this.props;
+
+    onClose();
+  }
+
+  deleteAccount() {
+    const { password } = this.state;
+
+    this.setState({ passwordSubmitted: true });
+
+    fetch('/accounts/verify_password', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ password }),
+    }).then((response) => {
+      if (response.ok) {
+        return this.setState({
```

Comment:
Should this clear the `validationMessage` and / or `validationErrorDetails` from previous error(s)?